api/app/controllers/api/v1/analytics_controller.rb

```diff
@@ -0,0 +1,60 @@
+module Api
+  module V1
+    class AnalyticsController < ApplicationController
+
+      before_action :authorize_user
+
+      def show
+        scope = analytics_scope
+
+        interaction_outcome = case scope
+        when nil
+          Analytics::Reports::Global.run(allow_cached_result: false, **analytics_params)
+        else
+          reporter_class = "Analytics::Reports::For#{scope.class.name}".constantize
+          reporter_class.run scope: scope, **analytics_params.to_h.symbolize_keys
+        end
+
+        render_jsonapi(interaction_outcome.result, serializer: ::V1::AnalyticsResultSerializer)
+      end
+
+      def create
+        outcome = Analytics::RecordCustomEvent.run custom_event_params.to_h.merge(user: current_user)
+        head(outcome.valid? ? 200 : 400)
```

Comment:
I want to make sure your responses fit in with the kind of response we expect from the API. See, for example, the `render_error_response` method or `respond_with_forbidden` in the application controller for examples.